modules/transport/Transport.js

```diff
@@ -0,0 +1,239 @@
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+
+/**
+ * Stores the currnet transport that have to be used.
+ */
+export default class Transport {
+    /**
+     * Creates new instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor({ transport } = {}) {
+        this._requestID = 0;
+
+        this._responseHandlers = new Map();
+
+        this._listeners = new Map();
+
+        this._unprocessedMessages = new Set();
+
+        this.addListener = this.on;
+
+        if (transport) {
+            this.setTransport(transport);
+        }
+    }
+
+    /**
+     * Disposes the current transport.
+     *
+     * @returns {void}
+     */
+    _disposeTransport() {
+        if (this._transport) {
+            this._transport.dispose();
+            this._transport = null;
```

Comment:
- `_disposeTransport` -> `_disposeBackend` because we're not disposing the `Transport` but its `TransportBackend`
- `_transport` -> `_backend`

We're in `Transport` so `backend` is clear enough in the context and it's not necessary to state `transportBackend`.